x-pack/test/apm_api_integration/tests/services/throughput.ts

```diff
@@ -89,51 +89,98 @@ export default function ApiTest({ getService }: FtrProviderContext) {
+                  start: metadata.start,
+                  end: metadata.end,
+                  transactionType: 'request',
+                  environment: 'ENVIRONMENT_ALL',
+                },
               },
-              query: {
-                kuery: 'processor.event : "transaction"',
-                start: metadata.start,
-                end: metadata.end,
-                transactionType: 'request',
-                environment: 'ENVIRONMENT_ALL',
-              },
-            },
+            });
+            throughputResponse = response.body;
+          });
+
+          it('returns some data', () => {
+            expect(throughputResponse.currentPeriod.length).to.be.greaterThan(0);
+            expect(throughputResponse.previousPeriod.length).not.to.be.greaterThan(0);
+
+            const nonNullDataPoints = throughputResponse.currentPeriod.filter(({ y }) =>
+              isFiniteNumber(y)
+            );
+
+            expect(nonNullDataPoints.length).to.be.greaterThan(0);
+          });
+
+          it('has the correct start date', () => {
+            expectSnapshot(
+              new Date(first(throughputResponse.currentPeriod)?.x ?? NaN).toISOString()
+            ).toMatchInline(`"2021-08-03T06:50:00.000Z"`);
+          });
+
+          it('has the correct first value', () => {
+            expectSnapshot(first(throughputResponse.currentPeriod)?.y).toMatchInline(`2`);
           });
-          throughputResponse = response.body;
-        });
 
-        it('has the correct throughput in tps', async () => {
-          const avgTps = mean(throughputResponse.currentPeriod.map((d) => d.y));
-          expectSnapshot(avgTps).toMatchInline(`0.124043715846995`);
-          expectSnapshot(throughputResponse.throughputUnit).toMatchInline(`"second"`);
+          it('has the correct end date', () => {
+            expectSnapshot(
+              new Date(last(throughputResponse.currentPeriod)?.x ?? NaN).toISOString()
+            ).toMatchInline(`"2021-08-03T07:20:00.000Z"`);
+          });
 
-          // this tpm value must be similar tp tpm value calculated in the previous spec where metric docs were used
-          const avgTpm = avgTps * 60;
-          expectSnapshot(avgTpm).toMatchInline(`7.44262295081967`);
+          it('has the correct last value', () => {
+            expectSnapshot(last(throughputResponse.currentPeriod)?.y).toMatchInline(`0`);
+          });
+
+          it('has the correct number of buckets', () => {
+            expectSnapshot(throughputResponse.currentPeriod.length).toMatchInline(`61`);
+          });
+
+          it('has the correct throughput avg', () => {
+            const avg = mean(throughputResponse.currentPeriod.map((d) => d.y));
+            expectSnapshot(avg).toMatchInline(`7.44262295081967`);
+          });
+
+          it('has the correct throughput', () => {
+            expectSnapshot(throughputResponse.currentPeriod).toMatch();
+          });
         });
       });
     }
```

Comment:
Will you update this test to use the new (synthbeans-node) archive?

Reply:
working on that 👷🏻‍♂️